src/components/messages/MsgDetails/index.tsx

```diff
@@ -0,0 +1,121 @@
+import { InfoDetails } from '@/components/transactions/InfoDetails'
+import { TxDataRow } from '@/components/transactions/TxDetails/Summary/TxDataRow'
+import type { Message } from '@/hooks/useMessages'
+
+import txDetailsCss from '@/components/transactions/TxDetails/styles.module.css'
+import singleTxDecodedCss from '@/components/transactions/TxDetails/TxData/DecodedData/SingleTxDecoded/styles.module.css'
+import infoDetailsCss from '@/components/transactions/InfoDetails/styles.module.css'
+import useWallet from '@/hooks/wallets/useWallet'
+
+const Msg = ({ message }: { message: Message['message'] }): ReactElement => {
+  const [showMsg, setShowMsg] = useState(true)
+
+  const handleToggleMsg = () => {
+    setShowMsg((prev) => !prev)
+  }
+
+  if (typeof message === 'string') {
+    return <>{message}</>
+  }
+
+  return (
+    <div>
+      {showMsg && (
+        <pre style={{ margin: 0 }}>
+          <code>{JSON.stringify(message, null, 2)}</code>
+        </pre>
+      )}
+      <Link
+        component="button"
+        onClick={handleToggleMsg}
+        fontSize="medium"
+        fontWeight={700}
+        sx={{ textDecoration: 'underline' }}
+      >
+        {showMsg ? 'Hide' : 'Show'}
+      </Link>
+    </div>
+  )
+}
+
+const MsgDetails = ({ msg }: { msg: Message }) => {
+  const wallet = useWallet()
+
+  return (
+    <div className={txDetailsCss.container}>
+      <div className={txDetailsCss.details}>
+        <div className={txDetailsCss.txData}>
+          <InfoDetails title="Created by:">
+            <EthHashInfo
+              address={msg.proposedBy.value || ''}
+              name={msg.proposedBy.name}
+              customAvatar={msg.proposedBy.logoUri}
+              shortAddress={false}
+              showCopyButton
+              hasExplorer
+            />
+          </InfoDetails>
+        </div>
+
+        <div className={txDetailsCss.txSummary}>
+          <TxDataRow title="Message:">
+            <Msg message={msg.message} />
+          </TxDataRow>
+          <TxDataRow title="Hash:">{msg.messageHash}</TxDataRow>
+          <TxDataRow title="Created:">{formatDateTime(msg.creationTimestamp)}</TxDataRow>
+          <TxDataRow title="Last modified:">{formatDateTime(msg.modifiedTimestamp)}</TxDataRow>
+        </div>
+
+        {msg.preparedSignature && (
+          <div className={classNames(txDetailsCss.txSummary, txDetailsCss.multiSend)}>
+            <TxDataRow title="Prepared signature:">{msg.preparedSignature}</TxDataRow>
+          </div>
+        )}
+
+        <div className={txDetailsCss.multiSend}>
+          {msg.confirmations.map((confirmation, i) => (
+            <Accordion
+              variant="elevation"
+              key={confirmation.signature}
+              defaultExpanded={confirmation.owner.value === wallet?.address}
+            >
+              <AccordionSummary expandIcon={<ExpandMoreIcon />}>
+                <div className={singleTxDecodedCss.summary}>
+                  <CodeIcon />
```

Comment:
IMO we should use a different Icon here as it's not multisend calls / functions but confirmations.
Probably a checkmark or something like that.

Reply:
The design uses a `CodeIcon` but I tend to agree. cc @liliiaorlenko